lib/fluent/diagtool/diagutils.rb

```diff
@@ -14,15 +14,28 @@
 #    limitations under the License.
 #
 
+module Diagtool
+  ON_WINDOWS = /mingw/.match?(RUBY_PLATFORM)
+end
+
 require 'logger'
 require 'fileutils'
 require 'fluent/diagtool/collectutils'
 require 'fluent/diagtool/maskutils'
 require 'fluent/diagtool/validutils'
+require 'fluent/diagtool/windows/diagutils' if Diagtool::ON_WINDOWS
```

Comment:
If considering to make it more abstract implementation, 

`require 'fluent/diagtool/diagutils/windows'` or `require 'fluent/diagtool/diagutils/linux'` seems reasonable hierarchy.

Reply:
Sorry, could you please tell me a little more about this?

I think, ideally, directly under `fluent/diagtool/` should be an abstraction and common logic, and platform-specific logic should be under `fluent/diagtool/windows/` or `fluent/diagtool/unix-like/`.
(For now, existing logic remains to reduce the impact)

Reply:
I've focused on diagutils and derivatives so I felt that 
diagutils.rb and diagutils/windows.rb or something is better.

In the contrast, if focused on diagtool, diagtool/diagutils.rb and diagtool/windows/diagutils.rb may be reasonable.

Thus, it depends on what we focused on.

For the future improvements, I've reconsidered the latter is better.
Sorry for confusing

Reply:
I see! Thanks for your feedback.
Indeed, it depends on what we are focused on.

Reply:
This structure can be easily changed, so if we need it later, let's do so!